lib/document.js

```diff
@@ -2455,13 +2457,23 @@ Document.prototype.$__validate = function(pathsToValidate, options, callback) {
     pathDetails[0].filter((path) => this.isModified(path)) :
     pathDetails[0];
   const skipSchemaValidators = pathDetails[1];
-
   if (typeof pathsToValidate === 'string') {
     pathsToValidate = pathsToValidate.split(' ');
   }
   if (Array.isArray(pathsToValidate)) {
     paths = _handlePathsToValidate(paths, pathsToValidate);
   }
+  else if (pathsToSkip) {
+    // make a new variable that is paths intersection !options.pathsToSkip
+    const valPaths = new Set(paths);
+    for (const path of pathsToSkip) {
+      if (valPaths.has(path)) {
+        valPaths.delete(path);
+      }
+    }
+    // then send that to be validated
+    paths = _handlePathsToSkip(paths, valPaths);
```

Comment:
Why is the separate `valPaths` set here? Why not just pass `pathsToSkip` to `_handlePathsToSkip()`?